### PR TITLE
Support macOS arm64 build

### PR DESCRIPTION
**Pull request by mbautin**
Support arm64 (Apple Silicon) build on macOS. Because it is also possible to build for x86_64 on an Apple Silicon macOS machine, we are using the environment variable YB_TARGET_ARCH to specify the target architecture. To force build scripts to run with the correct architecture, we use the `arch` command as well as a Homebrew installation in /opt/homebrew.

### CODE REVIEW

File: yb-thirdparty-common.sh [hunk 1]
```diff
@@ -107,6 +107,42 @@ activate_virtualenv() {
...
 }
 
+# Re-executes the current script with the correct macOS architecture.
+ensure_correct_mac_architecture() {
+  if [[ $OSTYPE != darwin* ]]; then
+    return
+  fi
+  if [[ -z ${YB_TARGET_ARCH:-} ]]; then
+    local uname_output
+    uname_output=$( uname -a )
```
Comment (svarnau) [inline yb-thirdparty-common.sh:117]: Should we check more specific option instead of the "all" option?  uname -p ?

**Review by svarnau — COMMENTED** [review]
Couple questions, but looks okay.

File: python/build_definitions/cassandra_cpp_driver.py [hunk 2]
```diff
@@ -23,8 +23,8 @@
 class CassandraCppDriverDependency(Dependency):
     def __init__(self) -> None:
         super(CassandraCppDriverDependency, self).__init__(
-                'cassandra-cpp-driver', '2.9.0-yb-12',
-                'https://github.com/YugaByte/cassandra-cpp-driver/archive/{0}.tar.gz',
+                'cassandra-cpp-driver', '2.9.0-yb-12-mb-2',
+                'https://github.com/mbautin/cassandra-cpp-driver/archive/{0}.tar.gz',
                 BUILD_GROUP_INSTRUMENTED)
```
[the remark this answers is not in the document]
Comment (mbautin) [reply python/build_definitions/cassandra_cpp_driver.py:28]: Yes, need to merge https://github.com/yugabyte/cassandra-cpp-driver/pull/12 first

[the remark this answers is not in the document]
Comment (mbautin) [reply python/build_definitions/cassandra_cpp_driver.py:28]: Updated.

File: python/yugabyte_db_thirdparty/arch.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+from yugabyte_db_thirdparty.util import add_path_entry
+
+
+g_target_arch: Optional[str] = None
+
+MACOS_CPU_ARCHITECTURES = ['x86_64', 'arm64']
+HOMEBREW_BIN_DIR_BY_ARCH = {
+    'x86_64': '/usr/local/bin',
+    'arm64': '/opt/homebrew/bin'
```
[the remark this answers is not in the document]
Comment (mbautin) [reply python/yugabyte_db_thirdparty/arch.py:28]: Updated to use `uname -p`.

File: yb-thirdparty-common.sh (hunk 1 above)
Comment (mbautin) [reply yb-thirdparty-common.sh]: Updated to use `uname -p`.

**Review by svarnau — APPROVED** [review]
LGTM

**Pull request merged**